spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcValueWriters.java

```diff
@@ -73,6 +80,16 @@ public void nonNullWrite(int rowId, UTF8String data, ColumnVector output) {
     }
   }
 
+  private static class UUIDWriter implements OrcValueWriter<UTF8String> {
+    private static final UUIDWriter INSTANCE = new UUIDWriter();
+
+    @Override
+    public void nonNullWrite(int rowId, UTF8String data, ColumnVector output) {
+      ByteBuffer buffer = UUIDUtil.convertToByteBuffer(UUID.fromString(data.toString()));
```

Comment:
This allocates a buffer. We may want to have a buffer here as a thread-local or a field to avoid allocation in a tight loop.

Reply:
Probably worth mentioning in a comment?

Reply:
makes sense, I've added a comment to this as part of https://github.com/apache/iceberg/pull/7496